components/causal_ts/src/lib.rs

```diff
@@ -31,22 +30,22 @@ pub trait CausalTsProvider: Send + Sync {
         block_on(self.async_get_ts())
     }
 
-    /// Flush (cached) timestamps to keep causality on some events, such as
-    /// "leader transfer".
-    fn flush(&self) -> Result<()> {
+    /// Flush (cached) timestamps and return first timestamp to keep causality
+    /// on some events, such as "leader transfer".
+    fn flush(&self) -> Result<TimeStamp> {
```

Comment:
Is it still used?

Reply:
backup component need it, ref [here](https://github.com/tikv/tikv/blob/master/components/backup/src/endpoint.rs#L985).

Reply:
How about inlining  the implementation? So users may not misuse the function.

Reply:
Why `#[inline]` can prevent users from misusing？
I try to understand that call `block_on` in a async func is not illegal, but call `block_on` in a sync func is illegal even if the sync func is called in a async func. So we can use `#[inline]`  to prevent users misusing. Am I understanding that right?

Reply:
Based on the above, I try to add `#[inline(always)]` for `get_ts`, and call `get_ts` in [a async func](https://github.com/tikv/tikv/blob/master/components/cdc/src/endpoint.rs#L1028), but I don't get error. Now, I am confused.

Reply:
What I mean is calling `block_on(async_flush())` instead of calling `flush`, so that `flush` can be removed.

Reply:
Allright, I removed it, thanks!